### ISSUE TITLE
Changing HTTP Method from GET to POST for connect/authorize and connect/endsession: Is It Possible?

### ISSUE THREAD

**Issue by abdrizalf12** [authts/oidc-client-ts#1419]
Is there a way to switch the method from GET to POST when performing connect/authorize and connect/endsession operations?

Are there any alternatives to using oidc-client.ts or oidc-client.js for this purpose ?

**Comment by pamapa** [comment]
no, the request functions use what is defined by the standard OAuth2.0

**Comment by abdrizalf12** [comment]
Hi @pamapa , Thank you for the response. We understand that changing the end session request to a POST method isn't feasible due to adherence to the OAuth 2.0 standard.

Our audit team has highlighted concerns regarding the **id_token_hint** parameter in the **connect/endsession** API call could potentially expose us to session fixation or hijacking attacks. Such attacks could allow an unauthorized user to impersonate a legitimate user and gain illegitimate access to the web application.

We recognize that one workaround could be to avoid requesting the "openid" scope. However, this is not a viable solution for our system, which requires this scope.

Could you please confirm whether using the GET method for the **connect/endsession** API call, as per the standard, with the id_token_hint as query parameter, could indeed be susceptible to hacking and be considered a security vulnerability?

**Comment by pamapa** [comment]
> Could you please confirm whether using the GET method for the **connect/endsession** API call, as per the standard, with the id_token_hint as query parameter, could indeed be susceptible to hacking and be considered a security vulnerability?

I will not give you legal advise. This library strictly follows the OAuth2.0 Standard. This end session can be done with `signoutRedirect`, after https://openid.net/specs/openid-connect-rpinitiated-1_0.html#RPLogout, GET or POST must be supported on the IdP side. The redirect is done via `_redirectNavigator`, which could be changed via `redirectNavigator`. In order to do so you must read through the source code...

**Comment by etremblay** [comment]
Could it be acceptable if it were an opt-in option ?

We could try to contribute a pull request for this.
